azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultProcessorHelper.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.keyvault.spring;
+
+import com.microsoft.azure.keyvault.KeyVaultClient;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MutablePropertySources;
+import org.springframework.core.env.StandardEnvironment;
+
+public class KeyVaultProcessorHelper {
+    private ConfigurableEnvironment environment;
+    private String id;
+    private String key;
+    private String uri;
+    private long timeout;
+
+    public KeyVaultProcessorHelper(ConfigurableEnvironment environment,
+                                   String id, String key, String uri, long timeout) {
+        this.environment = environment;
+        this.id = id;
+        this.key = key;
+        this.uri = uri;
+        this.timeout = timeout;
+    }
+
+    public void addKeyVaultPropertySource() {
```

Comment:
You can extract the properties here since you have the `Environment`. That makes sure the logic is in one place.